src/wayland/shell/xdg/mod.rs

```diff
@@ -189,18 +210,59 @@ macro_rules! xdg_role {
 
                 Some(configure.into())
             }
+
+            /// Gets the latest state that has been configured
+            /// on the server. The state can include changes
+            /// that have been made on the server but not yet
+            /// acked or committed by the client. It does not
+            /// include the current pending state.
+            ///
+            /// This can be used for example to check if the
+            /// pending state is different from the last configured.
+            fn current_server_state(&self) -> &$state {
```

Comment:
Given the documentation on this method, I guess it is intended to be `pub` ?

If so, please split the first sentence on its own line with an empty line, to ensure rustdoc renders it nicely.

Reply:
I was not sure if the two functions should be `pub` but added the doc anyway. If they provide value i am happy to make the pub and fix the small doc issue.

Reply:
It seems that this could be useful to have this method public, yes.

Reply:
The two methods are now public and i changed the docs a bit. Additionally i found two doc issues which i fixed.
The link to `send_configure` was wrong and the link to the `xdg` module in `shell` was broken.